### PR TITLE
Disable per-session thread pool for web

### PR DESCRIPTION
**Pull request by snnn**
### Description
ORT web prefers to use a global thread pool for all inference sessions.  See how OrtCreateSession is implemented in https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/wasm/api.cc#L183 .

Application code can only the global thread poo. However, internal testing code still often use per-session threadpool. This pr is to fix the inconsistency. 

### Motivation and Context
#18476

### CODE REVIEW

**Comment by fs-eire** [comment]
this may apply to all wasm build, including:
- single thread build ( will this be a problem? )
- static wasm library
- ort-wasm for onnxruntime-web ( this is already enabled global threadpool , but only when PThread is defined:
  ```
  OrtSession* OrtCreateSession(void* data, size_t data_length, OrtSessionOptions* session_options) {
  #if defined(__EMSCRIPTEN_PTHREADS__)
    RETURN_NULLPTR_IF_ERROR(DisablePerSessionThreads, session_options);
  ```

**Comment by snnn** [comment]
> single thread build ( will this be a problem? )

No. Because there won't be a thread pool in the global env in a single thread build.

File: onnxruntime/core/framework/session_options.h [hunk 1]
```diff
@@ -111,7 +111,11 @@ struct SessionOptions {
 
   // By default the session uses its own set of threadpools, unless this is set to false.
   // Use this in conjunction with the CreateEnvWithGlobalThreadPools API.
+#if defined(__wasm__)
```
Comment (snnn) [inline onnxruntime/core/framework/session_options.h:114]: @fs-eire do you suggest adding `&& __EMSCRIPTEN_PTHREADS__` here?

Comment (fs-eire) [reply onnxruntime/core/framework/session_options.h:114]: yes

File: onnxruntime/test/framework/inference_session_test.cc [hunk 2]
```diff
@@ -837,6 +840,9 @@ TEST(InferenceSessionTests, PreAllocateOutputVector) {
 }
 
 TEST(InferenceSessionTests, ConfigureVerbosityLevel) {
+  if constexpr (!SessionOptions::DEFAULT_USE_PER_SESSION_THREADS) {
+    GTEST_SKIP() << "Skipping the test";
```
Comment (fs-eire) [inline onnxruntime/test/framework/inference_session_test.cc:844]: does GTEST_SKIP() macro `return` internally?

**Comment by fs-eire** [comment]
It looks like some tests are not related to multi-thread, but they still get disabled. What is the reason behind that?

File: onnxruntime/test/framework/inference_session_test.cc (hunk 2 above)
Comment (snnn) [reply onnxruntime/test/framework/inference_session_test.cc:844]: Yes. GTest's  official doc has the following example:
```c++
TEST(SkipTest, DoesSkip) {
  GTEST_SKIP() << "Skipping single test";
  EXPECT_EQ(0, 1);  // Won't fail; it won't be executed
}
```

**Comment by snnn** [comment]
> It looks like some tests are not related to multi-thread, but they still get disabled. What is the reason behind that?

Normally we only create one OrtEnv per each process. The disabled tests are exceptions: they create a new OrtEnv inside of the test bodies. And since these OrtEnvs do not have a thread pool, creating a session that requires global thread pool would fail.

**Review by fs-eire — APPROVED** [review]

**Pull request merged**